pkgs/development/python-modules/cramjam/default.nix

```diff
@@ -13,29 +16,52 @@
 
 buildPythonPackage rec {
   pname = "cramjam";
-  version = "2.8.3";
+  version = "2.9.1";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "milesgranger";
-    repo = "pyrus-cramjam";
+    repo = "cramjam";
     tag = "v${version}";
-    hash = "sha256-1KD5/oZjfdXav1ZByQoyyiDSzbmY4VJsSJg/FtUFdDE=";
+    hash = "sha256-s+dKoZftI+aXcie1quAo/Xmlt1BQrzoCjwesDiaNABY=";
   };
 
+  postPatch =
+    # Unclear why dynamic library loading is not the default.
+    #
+    # error: could not find native static library `isal`, perhaps an -L flag is missing?
+    ''
+      substituteInPlace "$cargoDepsCopy"/isal-sys-*/build.rs \
+        --replace-fail "cargo:rustc-link-lib={}" "cargo:rustc-link-lib=dylib={}"
+    ''
+    # error: could not find native static library `blosc2`, perhaps an -L flag is missing?
+    + ''
+      substituteInPlace "$cargoDepsCopy"/blosc2-sys-*/build.rs \
+        --replace-fail "cargo:rustc-link-lib=blosc2" "cargo:rustc-link-lib=dylib=blosc2"
+    '';
+
   cargoDeps = rustPlatform.fetchCargoVendor {
-    inherit src;
-    hash = "sha256-Gpj/LUCx/ivYlmxNJnEZr8caEfV4OaQwEPNjz7vobsw=";
+    inherit pname src version;
+    hash = "sha256-xrqlVgAUTsqz7AOQdPCSxjRd/Z3ZVpr0qphw6qN6Wys=";
   };
 
-  buildAndTestSubdir = "cramjam-python";
-
   nativeBuildInputs = with rustPlatform; [
     cargoSetupHook
     maturinBuildHook
+    # Used by the build.rs script from blosc2-sys to find the blosc2 system library
+    pkg-config
   ];
 
-  buildInputs = lib.optional stdenv.hostPlatform.isDarwin libiconv;
+  buildInputs = [
+    c-blosc2
+    isa-l
+  ];
+
+  # Do not build those libraries from source. Use the nixpkgs ones instead.
+  maturinBuildFlags = [
+    "--features=use-system-blosc2-shared"
+    "--features=use-system-isal-shared"
```

Comment:
```suggestion
  ] ++ lib.optionals (stdenv.hostPlatform.system != "aarch64-darwin") [
    "--features=use-system-isal-shared"
```

Reply:
Hm, ISA shouldn't be enabled by default if I read the Cargo.toml correctly. The ISA features part of the `experimental` feature set, not the `default`: https://github.com/milesgranger/cramjam/blob/61564e7761e38e5ec55e7939ccd6a276c2c55d11/Cargo.toml#L17-L18

Reply:
As discussed in https://github.com/NixOS/nixpkgs/pull/378691, those are experimental/optional on master, so we could already disable them for the release version.